### PR TITLE
Activate several lints in regards to promises and async fn

### PR DESCRIPTION
**Pull request by thomaseizinger**
We've been bitten by bad usage of promises in the past (comparing them, not awaiting them, etc).

This PR activates several lints and fixes the respective problems. There are not that many but when reading through the [list of lints](https://palantir.github.io/tslint/rules/), several bugs crossed my mind that would have been caught by these lints.

This PR also inspired [a ticket on the SDK repo](https://github.com/comit-network/comit-js-sdk/issues/130), where we should likely do the same (probably even stricter).

### CODE REVIEW

File: api_tests/lib/actor.ts [hunk 1]
```diff
@@ -80,7 +80,7 @@ export class Actor {
         }
     }
 
-    public doComitAction(action: Action): Promise<Response> {
+    public async doComitAction(action: Action): Promise<Response> {
```
Comment (D4nte) [inline api_tests/lib/actor.ts]: Oh my god this is so nice, it detected that a `Promise` was returned and hence told you to add `async`?

File: api_tests/lib/bitcoind_instance.ts [hunk 2]
```diff
@@ -41,7 +41,7 @@ export class BitcoindInstance implements LedgerInstance {
         });
 
         this.process.on("exit", (code: number, signal: number) => {
-            console.log(`bitcoind exited with ${code || "signal " + signal}`);
+            console.log(`bitcoind exited with ${code || `signal ${signal}`}`);
```
Comment (D4nte) [inline api_tests/lib/bitcoind_instance.ts]: Hum, what is this lint rule?

**Comment by da-kami** [comment]
Like the idea, would be great to get https://github.com/comit-network/comit-rs/pull/2054 in before changing things in the `lib` though...

File: api_tests/lib/bitcoind_instance.ts (hunk 2 above)
Comment (thomaseizinger) [reply api_tests/lib/bitcoind_instance.ts]: Not using + on mixed types like numbers and strings.

File: api_tests/lib/actor.ts (hunk 1 above)
Comment (thomaseizinger) [reply api_tests/lib/actor.ts]: Yes :)

**Comment by thomaseizinger** [comment]
@da-kami yeah sure, I can easily rebase :)

**Comment by thomaseizinger** [comment]
I will take your comments as approvals and merge this @D4nte and @da-kami.

bors r+

**Comment by D4nte** [comment]
> I will take your comments as approvals and merge this @D4nte and @da-kami.
> 
> bors r+

Well, they are meant to be but aren't due to GitHub trickery!

**Pull request merged**